backend/tests/test_doc.py

```diff
@@ -44,10 +63,44 @@ def test_doc_delete(memory_session: Session, logged_in_client: TestClient):
 
     assert files < set(str(x) for x in settings.storage_path.glob("*"))
 
+    req = client.delete(f"/api/v1/documents/{document_id}/")
+    assert 400 <= req.status_code < 500
+
     req = logged_in_client.delete(f"/api/v1/documents/{document_id}/")
     assert req.status_code == 200
 
     for table in checked_tables:
         assert counts[table] == memory_session.query(table).count()
 
     assert files == set(str(x) for x in settings.storage_path.glob("*"))
+
+
+@pytest.mark.parametrize(
+    "method,url,need_specific_user",
+    [
+        ["get", "/api/v1/documents/", False],
+        ["get", "/api/v1/documents/{document_id}/", True],
+        ["delete", "/api/v1/documents/{document_id}/", True],
+        ["get", "/api/v1/documents/{document_id}/tasks/", True],
+    ],
+)
+def test_user_auth(
+    logged_in_client: TestClient,
+    logged_in_client_2: TestClient,
+    client: TestClient,
+    document: str,
+    method: str,
+    url: str,
+    need_specific_user: bool,
+):
+    req = getattr(client, method)(url.format(document_id=document))
+    assert 400 <= req.status_code < 500
+
+    req = getattr(logged_in_client_2, method)(url.format(document_id=document))
+    if need_specific_user:
+        assert 400 <= req.status_code < 500
+    else:
+        assert 200 <= req.status_code < 300
+
+    req = getattr(logged_in_client, method)(url.format(document_id=document))
```

Comment:
I would be happy about two comment like "try to acces with owning user" and "try to access with different user"